src/config.rs

```diff
@@ -28,6 +34,42 @@ impl FromStr for WorkMode {
     }
 }
 
+#[derive(Debug)]
+pub enum ConfigError {
+    Io(IoError, String),
+    TomlError(toml::ParserError, String),
+}
+
+impl Error for ConfigError {
+    fn description(&self) -> &str {
+        match self {
+            &ConfigError::Io(ref e, _) => e.description(),
+            &ConfigError::TomlError(ref s, _) => s.description(),
+        }
+    }
+}
+
+impl Display for ConfigError {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self {
+            &ConfigError::Io(ref e, ref filename) => {
+                write!(f, "\"{}\": ", filename);
+                e.fmt(f)
+            }
+            &ConfigError::TomlError(ref err, ref filename) => {
+                write!(f, "Toml parsing error on \"{}\": error: {}",
+                         filename, err.desc)
```

Comment:
After the issue with the location below is settled, `err.desc` would just be `err`